backend/app/api/v1/endpoints/teams.py

```diff
@@ -205,29 +199,29 @@ def get_team(
     Returns:
         TeamSideView: The requested team.
     """
-    if current_user.role in (Roles.ADMIN, Roles.BOARD, Roles.COACH, Roles.PLAYER):
-        if current_user.role == Roles.PLAYER:
-            if (team := current_user.player.team) is None:
-                raise MissingException(Team.__name__)
-        else:
-            team = get_team_by_id(team_id=team_id, db=db)
-        if current_user.role in (Roles.ADMIN, Roles.BOARD, Roles.PLAYER) or (
-            current_user.role == Roles.COACH and current_user.id == team.coach_id
-        ):
-            return TeamSideView(
-                id=team.id,
-                name=team.name,
-                players=[
-                    PlayerOnlyBaseInfo(
-                        user_id=player.user_id, user_full_name=player.user.full_name
-                    )
-                    for player in team.players
-                ],
-                coach=CoachOnlyBaseInfo(
-                    user_id=team.coach.user.id,
-                    user_full_name=team.coach.user.full_name,
+    if current_user.role == Roles.PLAYER:
+        if (team := current_user.player.team) is None:
+            raise MissingException(Team.__name__)
+    else:
+        team = get_team_by_id(team_id=team_id, db=db)
+
+    if current_user.role != Roles.COACH or (
+        current_user.role == Roles.COACH and current_user.id == team.coach_id
+    ):
+        return TeamSideView(
+            id=team.id,
+            name=team.name,
+            players=[
+                PlayerOnlyBaseInfo(
+                    user_id=player.user_id, user_full_name=player.user.full_name
                 )
-                if team.coach is not None
-                else None,
+                for player in team.players
+            ],
+            coach=CoachOnlyBaseInfo(
+                user_id=team.coach.user.id,
+                user_full_name=team.coach.user.full_name,
             )
+            if team.coach is not None
```

Comment:
o co cho

Reply:
Jeśli team ma coacha, to podajemy jego dane, jeśli nie, to podajemy None.